generators/app/templates/client/server.js

```diff
@@ -0,0 +1,22 @@
+require('./config/env');
+var path = require('path');
+var url = require('url');
+var proxy = require('proxy-middleware');
+var history = require('connect-history-api-fallback');
+var serveStatic = require('express').static;
+var server = require('express')();
+
+var port = process.env.NODE_ENV == 'production' ? process.env.PORT : 8080;
+
+server.use(history());
+server.use('/', serveStatic(path.join(__dirname, 'public')));
+
+if(process.env.NODE_ENV !== 'production') {
+  server.use('/', proxy(url.parse(process.env.WEBPACK_DEV_SERVER_URL)))
+}
+
+server.use('/', proxy(url.parse(process.env.API_URL)));
```

Comment:
Well, the generator was already set up to proxy everything from Rails to `/api`. Rails will not know it but by design. It can be made to know, if you want, I just thought it was better if inside Rails we could act as if the `/api` namespace didn't exist.

Also, the way it was setup before you overwrote with your version, `/assets` was the only route in express and, technically, was only a route in express on development. In production it just served everything under public, like expected. But in production, webpack generated the bundles inside `public/assets`.

Also the way it was setup before, everything under React Router was automatically namespaced to `/app`. The Links and the Routes could (and should) be written without taking the the `/app` in consideration and it was automatically added.